Python/optimizer.c

```diff
@@ -779,6 +800,76 @@ translate_bytecode_to_trace(
 #undef DPRINTF
 }
 
+static int
+remove_unneeded_uops(_PyUOpInstruction *trace, int trace_length)
+{
+    // Stage 1: Replace unneeded SAVE_IP uops with NOP.
+    // Note that we don't enter stubs, those SAVE_IPs are needed.
+    int last_save_ip = -1;
```

Comment:
I think it might be clearer if we call this variable `has_previous_save_ip` and make it a boolean.

Reply:
That would imply it's boolean, but it's an index.

Reply:
:facepalm: I thought I only saw it being used as a boolean, but then saw it's used to index the trace. Yeah sorry my mistake!